src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportDestinationClient/AzureExportDestinationClient.cs

```diff
@@ -0,0 +1,126 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.Export.ExportDestinationClient
+{
+    public class AzureExportDestinationClient : IExportDestinationClient
+    {
+        private CloudBlobClient _blobClient = null;
+        private CloudBlobContainer _blobContainer = null;
+
+        private Dictionary<Uri, CloudBlockBlobWrapper> _uriToBlobMapping = new Dictionary<Uri, CloudBlockBlobWrapper>();
+        private Dictionary<(Uri FileUri, uint PartId), Stream> _streamMappings = new Dictionary<(Uri FileUri, uint PartId), Stream>();
+
+        public string DestinationType => "azure-block-blob";
+
+        public async Task ConnectAsync(string connectionSettings, CancellationToken cancellationToken, string containerId = null)
+        {
+            EnsureArg.IsNotNullOrWhiteSpace(connectionSettings, nameof(connectionSettings));
+
+            string decodedConnectionString;
```

Comment:
nit: This could happen at the layer above since every client must implement the base64 decoding. It can be decoded before it's even stored in keyvault.

Reply:
(doesn't have to be in this PR.)